### PR TITLE
big gradle normalization

### PR DESCRIPTION
**Pull request by zarsky-broad**
When reviewing, I recommend clicking file filter and hiding the lockfiles.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,30 +28,39 @@ subprojects { subproject ->
...
 
+    configurations {
+        testImplementation.extendsFrom testFixturesImplementation
+    }
+
     dependencies {
-        api 'com.google.auth:google-auth-library-oauth2-http:0.20.0'
-        api 'com.google.cloud:google-cloud-core-http:1.93.5'
+        if (subproject.name != 'common') {
+            api project(':common')
```
Comment (wchamber) [inline build.gradle:37]: there's also the cloud-resource-schema which should not include 'common'. I'm a little worried about introducing circular dependencies here - we have to be careful that we exclude anything that 'common' depends on. I don't think the alternative of explicitly depending on common in each of the subprojects that requires 'common' is that bad.

File: common/src/main/java/bio/terra/cloudres/common/cleanup/CleanupRecorder.java [hunk 2]
```diff
@@ -100,7 +99,7 @@ private static void publish(CloudResourceUid resource, ClientConfig clientConfig
     ByteString data;
     try {
       data = ByteString.copyFromUtf8(objectMapper.writeValueAsString(body));
-    } catch (JsonProcessingException e) {
+    } catch (IOException e) {
```
Comment (wchamber) [inline common/src/main/java/bio/terra/cloudres/common/cleanup/CleanupRecorder.java:102]: why is this being changed? As is, we're explicitly wrapping the JsonProcessingException while allowing other IOExceptions to propagate. With this change, we're considering any IOException to be a serialization error, which is probably not true.
This is also a change to the 'common' package which should cause us to bump our package version.

**Review by wchamber — CHANGES_REQUESTED** [review]
One of the "features" of how the versioning was set up in the `gradle.properties` files is that we could version the packages independently and advance them only when they changed. I don't think that's the case anymore? Any thoughts on how we can continue to do that?

File: common/src/main/java/bio/terra/cloudres/common/cleanup/CleanupRecorder.java (hunk 2 above)
Comment (wchamber) [reply common/src/main/java/bio/terra/cloudres/common/cleanup/CleanupRecorder.java:102]: I retract my objection, sorry, didn't look closely.
I do think this requires our version bumping, which would probably be better to do in a separate PR.

Comment (zarsky-broad) [reply common/src/main/java/bio/terra/cloudres/common/cleanup/CleanupRecorder.java:102]: It wouldn't compile for me without this change, so I think that should probably happen first?

**Review by wchamber — APPROVED** [review]
thanks!

File: common/build.gradle [hunk 3]
```diff
@@ -1,15 +1,17 @@
+version = '0.1.0'
```
Comment (wchamber) [inline common/build.gradle:1]: let's bump the version of this to 0.1.1 since we are making a code change

**Review by wchamber — APPROVED** [review]

**Pull request merged**